File: api/functional/flow-creator/src/main/java/com/emanuelvictor/api/functional/flowcreator/domain/services/ChoiceService.java
package com.emanuelvictor.api.functional.flowcreator.domain.services;

import com.emanuelvictor.api.functional.flowcreator.domain.entities.Choice;
import com.emanuelvictor.api.functional.flowcreator.domain.entities.alternative.IntermediaryAlternative;
import com.emanuelvictor.api.functional.flowcreator.domain.ports.repositories.ChoiceRepository;
import lombok.RequiredArgsConstructor;

/**
 *
 */
@RequiredArgsConstructor
public class ChoiceService {

    private final ChoiceRepository choiceRepository;

    /**
     * @param alternative {@link IntermediaryAlternative}
     * @return {@link Choice}
     */
    public Choice makeChoice(final IntermediaryAlternative alternative) {

        return choiceRepository.save(new Choice(alternative));
    }


}

File: api/functional/flow-creator/src/main/kotlin/com/emanuelvictor/api/functional/flowcreator/domain/services/ChoiceService.kt
package com.emanuelvictor.api.functional.flowcreator.domain.services

import com.emanuelvictor.api.functional.flowcreator.domain.entities.alternative.IntermediaryAlternative
import com.emanuelvictor.api.functional.flowcreator.domain.entities.choice.Choice
import com.emanuelvictor.api.functional.flowcreator.domain.services.generic.AbstractService
import org.springframework.data.domain.Page
import org.springframework.data.domain.PageImpl
import org.springframework.data.domain.Pageable

/**
 *
 */
class ChoiceService : AbstractService<Choice, Long>() {

    /**
     * @param alternative [IntermediaryAlternative]
     * @return [Choice]
     */
    fun makeChoice(alternative: IntermediaryAlternative): Choice {
        return repository.save(Choice(alternative))
    }

    /**
     *
     */
    fun listByFilters(defaultFilter: String?, pageable: Pageable): Page<Choice> {
        return PageImpl(repository.findAll().toList())
    }
}